airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/attribution_report.py

```diff
@@ -0,0 +1,172 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import pendulum
+import requests
+from source_amazon_ads.schemas import AttributionReportModel, Profile
+from source_amazon_ads.streams.common import AmazonAdsStream
+
+BRAND_REFERRAL_BONUS = "brb_bonus_amount"
+
+METRICS_MAP = {
+    "PERFORMANCE": [
+        "Click-throughs",
+        "attributedDetailPageViewsClicks14d",
+        "attributedAddToCartClicks14d",
+        "attributedPurchases14d",
+        "unitsSold14d",
+        "attributedSales14d",
+        "attributedTotalDetailPageViewsClicks14d",
+        "attributedTotalAddToCartClicks14d",
+        "attributedTotalPurchases14d",
+        "totalUnitsSold14d",
+        "totalAttributedSales14d",
+    ],
+    "PRODUCTS": [
+        "attributedDetailPageViewsClicks14d",
+        "attributedAddToCartClicks14d",
+        "attributedPurchases14d",
+        "unitsSold14d",
+        "attributedSales14d",
+        "brandHaloDetailPageViewsClicks14d",
+        "brandHaloAttributedAddToCartClicks14d",
+        "brandHaloAttributedPurchases14d",
+        "brandHaloUnitsSold14d",
+        "brandHaloAttributedSales14d",
+        "attributedNewToBrandPurchases14d",
+        "attributedNewToBrandUnitsSold14d",
+        "attributedNewToBrandSales14d",
+        "brandHaloNewToBrandPurchases14d",
+        "brandHaloNewToBrandUnitsSold14d",
+        "brandHaloNewToBrandSales14d",
+    ],
+}
+
+
+class AttributionReport(AmazonAdsStream):
+    """
+    This stream corresponds to Amazon Advertising API - Attribution Reports
+    https://advertising.amazon.com/API/docs/en-us/amazon-attribution-prod-3p/#/
+    """
+
+    model = AttributionReportModel
+    primary_key = None
+    data_field = "reports"
+    page_size = 300
+
+    report_type = ""
+    metrics = ""
+    group_by = ""
+
+    _next_page_token_field = "cursorId"
+    _current_profile_id = ""
+
+    REPORT_DATE_FORMAT = "YYYYMMDD"
+    CONFIG_DATE_FORMAT = "YYYY-MM-DD"
+    REPORTING_PERIOD = 90
+
+    def __init__(self, config: Mapping[str, Any], *args, **kwargs):
+        self._start_date = config.get("start_date")
+        self._req_start_date = ""
+        self._req_end_date = ""
+
+        super().__init__(config, *args, **kwargs)
+
+    def _set_dates(self, profile: Profile):
+        new_start_date = pendulum.now(tz=profile.timezone).subtract(days=1).date()
+        new_end_date = pendulum.now(tz=profile.timezone).date()
+
+        if self._start_date:
+            new_start_date = max(self._start_date, new_end_date.subtract(days=self.REPORTING_PERIOD))
+
+        self._req_start_date = new_start_date.format(self.REPORT_DATE_FORMAT)
+        self._req_end_date = new_end_date.format(self.REPORT_DATE_FORMAT)
+
+    @property
+    def http_method(self) -> str:
+        return "POST"
+
+    def read_records(self, *args, **kvargs) -> Iterable[Mapping[str, Any]]:
```

Comment:
I finally solved my local issue by adding a few profile Ids, though our dev account doesn't have any attribute reports. So, earlier, under the `secrets/spec.json`, when the profiles is null or empty, it continues to fetch. Can you add an if-statement around the line  to skip the read_records if `_profiles` is empty?

Reply:
current implementation

```
    def read_records(self, *args, **kvargs) -> Iterable[Mapping[str, Any]]:
        """
        Iterate through self._profiles list and send read all records for each profile.
        """
        for profile in self._profiles:
            self._set_dates(profile)
            self._current_profile_id = profile.profileId
            yield from super().read_records(*args, **kvargs)
```

if `self._profiles` is empty, it should not trigger any fetches

Reply:
can you do me a favor? Can you unset the `profiles` attribute in secrets/config.json, see if you get an error by running the `read` command? 

`python main.py read --config secrets/config.json --catalog integration_tests/configured_catalog.json`

```
{
  "client_id": "amzn1.application-oa2-client.client-id",
  "client_secret": "secret",
  "scope": "advertising::campaign_management",
  "refresh_token": "a very long token",
  "region": "NA"
}
```

Reply:
No error when `profiles` attribute is removed from `config.json` file.

When there are no user provided profiles, API call is made to fetch all profiles.

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/profiles.py#L33-L48

Reply:
thanks for the feedback, that's really helpful. 

The acceptance test is not passed. And the reason is that not all profiles are authorized to fetch reports, and the read command doesn't fail gracefully, which blocks the read on other streams. 

I read the codes again, looks like a profile is not independent of the others. In our dev test account, there are about 10 profiles fetched from the API, and only 4 of them are authorized to read the attribute reports. 

If I pass just these 4 profiles, the tests are good. But if I include another, the read command fails on that profile and stops fetching reports on other profiles. And under the `configured_catalog.json`, if I shuffle the `streams` collection to fetch attribute reports first, the following streams will stop. 

Similarly, if I pass a collection of profiles, but one of the profiles is authorized, the same error is thrown and not handled gracefully. 

Under a `read` command, we want to fetch as much data as possible, and reports from one profile shouldn't impact another profile or other streams. Can you adjust the code to satisfy that? 

You probably can use a [stream slice](https://docs.airbyte.com/connector-development/cdk-python/stream-slices/#stream-slicing) to store profiles, and fetch reports for each profile in that slice. The GitHub connector provides [a good example in fetching users in an org](https://github.com/airbytehq/airbyte/blob/dd4b7be5ca81c48fa8e47c059efefcdad6d88147/airbyte-integrations/connectors/source-github/source_github/streams.py#L418-L419). 

Let me know if you have a question.

Reply:
Can you help me in providing a way to reproduce the error?

I tried running this command 
```
python main.py read --config secrets/config.json --catalog integration_tests/configured_catalog.json
```


`config.json`
```
{
  "region": "NA",
  "client_id": "amzn1.application-oa2-client.",
  "start_date": "2022-09-13",
  "client_secret": "secret",
  "refresh_token": "secret",
  "report_wait_timeout": 30,
  "report_generation_max_retries": 5,
  "profiles": [1, 2]
}
```

It's returning 0 records for all the streams except `profiles`
```
{"type": "LOG", "log": {"level": "INFO", "message": "Read 3 records from profiles stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing profiles"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_display_campaigns "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_display_campaigns stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_display_campaigns"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_display_campaigns 0:00:00.000228"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_display_ad_groups "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_display_ad_groups stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_display_ad_groups"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_display_product_ads "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_display_product_ads stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_display_product_ads"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_display_targetings "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_display_targetings stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_display_targetings"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_product_campaigns "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_product_campaigns stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_product_campaigns"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_campaigns 0:00:00.000159"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_product_ad_groups "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_product_ad_groups stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_product_ad_groups"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_ad_groups 0:00:00.000161\nSyncing stream sponsored_product_campaigns 0:00:00.000159"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_product_keywords "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_product_keywords stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_product_keywords"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_ad_groups 0:00:00.000161\nSyncing stream sponsored_product_campaigns 0:00:00.000159\nSyncing stream sponsored_product_keywords 0:00:00.000143"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_product_negative_keywords "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_product_negative_keywords stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_product_negative_keywords"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_ad_groups 0:00:00.000161\nSyncing stream sponsored_product_campaigns 0:00:00.000159\nSyncing stream sponsored_product_keywords 0:00:00.000143\nSyncing stream sponsored_product_negative_keywords 0:00:00.000143"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_product_ads "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_product_ads stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_product_ads"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_ad_groups 0:00:00.000161\nSyncing stream sponsored_product_ads 0:00:00.000361\nSyncing stream sponsored_product_campaigns 0:00:00.000159\nSyncing stream sponsored_product_keywords 0:00:00.000143\nSyncing stream sponsored_product_negative_keywords 0:00:00.000143"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_product_targetings "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_product_targetings stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_product_targetings"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_ad_groups 0:00:00.000161\nSyncing stream sponsored_product_ads 0:00:00.000361\nSyncing stream sponsored_product_campaigns 0:00:00.000159\nSyncing stream sponsored_product_keywords 0:00:00.000143\nSyncing stream sponsored_product_negative_keywords 0:00:00.000143\nSyncing stream sponsored_product_targetings 0:00:00.000144"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_brands_campaigns "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_brands_campaigns stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_brands_campaigns"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_brands_campaigns 0:00:00.000144\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_ad_groups 0:00:00.000161\nSyncing stream sponsored_product_ads 0:00:00.000361\nSyncing stream sponsored_product_campaigns 0:00:00.000159\nSyncing stream sponsored_product_keywords 0:00:00.000143\nSyncing stream sponsored_product_negative_keywords 0:00:00.000143\nSyncing stream sponsored_product_targetings 0:00:00.000144"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_brands_ad_groups "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_brands_ad_groups stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_brands_ad_groups"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_brands_ad_groups 0:00:00.000145\nSyncing stream sponsored_brands_campaigns 0:00:00.000144\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_ad_groups 0:00:00.000161\nSyncing stream sponsored_product_ads 0:00:00.000361\nSyncing stream sponsored_product_campaigns 0:00:00.000159\nSyncing stream sponsored_product_keywords 0:00:00.000143\nSyncing stream sponsored_product_negative_keywords 0:00:00.000143\nSyncing stream sponsored_product_targetings 0:00:00.000144"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: sponsored_brands_keywords "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from sponsored_brands_keywords stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing sponsored_brands_keywords"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_brands_ad_groups 0:00:00.000145\nSyncing stream sponsored_brands_campaigns 0:00:00.000144\nSyncing stream sponsored_brands_keywords 0:00:00.000170\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_ad_groups 0:00:00.000161\nSyncing stream sponsored_product_ads 0:00:00.000361\nSyncing stream sponsored_product_campaigns 0:00:00.000159\nSyncing stream sponsored_product_keywords 0:00:00.000143\nSyncing stream sponsored_product_negative_keywords 0:00:00.000143\nSyncing stream sponsored_product_targetings 0:00:00.000144"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: attribution_report_products "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from attribution_report_products stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing attribution_report_products"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream attribution_report_products 0:00:00.007100\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_brands_ad_groups 0:00:00.000145\nSyncing stream sponsored_brands_campaigns 0:00:00.000144\nSyncing stream sponsored_brands_keywords 0:00:00.000170\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_ad_groups 0:00:00.000161\nSyncing stream sponsored_product_ads 0:00:00.000361\nSyncing stream sponsored_product_campaigns 0:00:00.000159\nSyncing stream sponsored_product_keywords 0:00:00.000143\nSyncing stream sponsored_product_negative_keywords 0:00:00.000143\nSyncing stream sponsored_product_targetings 0:00:00.000144"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: attribution_report_performance_adgroup "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from attribution_report_performance_adgroup stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing attribution_report_performance_adgroup"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream attribution_report_performance_adgroup 0:00:00.000172\nSyncing stream attribution_report_products 0:00:00.007100\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_brands_ad_groups 0:00:00.000145\nSyncing stream sponsored_brands_campaigns 0:00:00.000144\nSyncing stream sponsored_brands_keywords 0:00:00.000170\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_ad_groups 0:00:00.000161\nSyncing stream sponsored_product_ads 0:00:00.000361\nSyncing stream sponsored_product_campaigns 0:00:00.000159\nSyncing stream sponsored_product_keywords 0:00:00.000143\nSyncing stream sponsored_product_negative_keywords 0:00:00.000143\nSyncing stream sponsored_product_targetings 0:00:00.000144"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: attribution_report_performance_campaign "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from attribution_report_performance_campaign stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing attribution_report_performance_campaign"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream attribution_report_performance_adgroup 0:00:00.000172\nSyncing stream attribution_report_performance_campaign 0:00:00.000294\nSyncing stream attribution_report_products 0:00:00.007100\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_brands_ad_groups 0:00:00.000145\nSyncing stream sponsored_brands_campaigns 0:00:00.000144\nSyncing stream sponsored_brands_keywords 0:00:00.000170\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_ad_groups 0:00:00.000161\nSyncing stream sponsored_product_ads 0:00:00.000361\nSyncing stream sponsored_product_campaigns 0:00:00.000159\nSyncing stream sponsored_product_keywords 0:00:00.000143\nSyncing stream sponsored_product_negative_keywords 0:00:00.000143\nSyncing stream sponsored_product_targetings 0:00:00.000144"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Syncing stream: attribution_report_performance_creative "}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 0 records from attribution_report_performance_creative stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing attribution_report_performance_creative"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceAmazonAds runtimes:\nSyncing stream attribution_report_performance_adgroup 0:00:00.000172\nSyncing stream attribution_report_performance_campaign 0:00:00.000294\nSyncing stream attribution_report_performance_creative 0:00:00.007223\nSyncing stream attribution_report_products 0:00:00.007100\nSyncing stream profiles 0:00:00.005063\nSyncing stream sponsored_brands_ad_groups 0:00:00.000145\nSyncing stream sponsored_brands_campaigns 0:00:00.000144\nSyncing stream sponsored_brands_keywords 0:00:00.000170\nSyncing stream sponsored_display_ad_groups 0:00:00.000135\nSyncing stream sponsored_display_campaigns 0:00:00.000228\nSyncing stream sponsored_display_product_ads 0:00:00.000272\nSyncing stream sponsored_display_targetings 0:00:00.000179\nSyncing stream sponsored_product_ad_groups 0:00:00.000161\nSyncing stream sponsored_product_ads 0:00:00.000361\nSyncing stream sponsored_product_campaigns 0:00:00.000159\nSyncing stream sponsored_product_keywords 0:00:00.000143\nSyncing stream sponsored_product_negative_keywords 0:00:00.000143\nSyncing stream sponsored_product_targetings 0:00:00.000144"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing SourceAmazonAds"}}
```

Reply:
Sure, be happy to!

firstly, check `integration_tests/configured_catalog.json`, if `profiles` is one of the streams, the output will include the profile detail. 
```
"Read 3 records from profiles stream"
```

looks like you have 3 profiles. 

I guess all 3 profiles are authorized to read attribute_reports_*. Can you create another profile that is not authorized and add it to the array of profiles in `secrets/config.json`? For our case, if one of profiles in the array is not authorized, we will get the error of `This profileID is not authorized to use Amazon Attribution`, and the read stops.

Reply:
Any ideas on why I am not getting any error with fake profile IDs in `config.json`

```
{
  "region": "NA",
  "client_id": "amzn1.application-oa2-client.",
  "start_date": "2022-09-13",
  "client_secret": "secret",
  "refresh_token": "secret",
  "report_wait_timeout": 30,
  "report_generation_max_retries": 5,
  "profiles": [1, 2]
}
```

Reply:
I am not sure, maybe on the api side, the validation handles it before the authorization module is run?